lazysnap/src/main/java/com/google/accompanist/lazysnap/SnappingFlingBehavior.kt

```diff
@@ -0,0 +1,689 @@
+@Composable
+fun rememberSnappingFlingBehavior(
+    lazyListState: LazyListState,
+    snapOffsetForItem: (layoutInfo: LazyListLayoutInfo, itemInfo: LazyListItemInfo) -> Int = SnapOffsets.Center,
+    maximumFlingDistance: (LazyListLayoutInfo) -> Int = SnappingFlingBehaviorDefaults.maximumFlingDistance,
+    decayAnimationSpec: DecayAnimationSpec<Float> = rememberSplineBasedDecay(),
+    snapAnimationSpec: AnimationSpec<Float> = SnappingFlingBehaviorDefaults.snapAnimationSpec,
+    @Px endContentPadding: Int = 0,
+): SnappingFlingBehavior = remember(
+    lazyListState,
+    snapOffsetForItem,
+    maximumFlingDistance,
+    decayAnimationSpec,
+    snapAnimationSpec,
+) {
+    SnappingFlingBehavior(
+        lazyListState = lazyListState,
+        snapOffsetForItem = snapOffsetForItem,
+        maximumFlingDistance = maximumFlingDistance,
+        decayAnimationSpec = decayAnimationSpec,
+        snapAnimationSpec = snapAnimationSpec,
+    )
+}.apply {
+    this.endContentPadding = endContentPadding
+}
+
+/**
+ * Contains a number of values which can be used for the `snapOffsetForItem` parameter on
+ * [rememberSnappingFlingBehavior] and [SnappingFlingBehavior].
+ */
+@Suppress("unused")
+object SnapOffsets {
+    /**
+     * Snap offset which results in the start edge of the item, snapping to the start scrolling
+     * edge of the lazy list.
+     */
+    val Start: (LazyListLayoutInfo, LazyListItemInfo) -> Int = { _, _ -> 0 }
+
+    /**
+     * Snap offset which results in the item snapping in the center of the scrolling viewport
+     * of the lazy list.
+     */
+    val Center: (LazyListLayoutInfo, LazyListItemInfo) -> Int = { layoutInfo, itemInfo ->
+        (layoutInfo.layoutSize - itemInfo.size) / 2
+    }
+
+    /**
+     * Snap offset which results in the end edge of the item, snapping to the end scrolling
+     * edge of the lazy list.
+     */
+    val End: (LazyListLayoutInfo, LazyListItemInfo) -> Int = { layoutInfo, itemInfo ->
+        layoutInfo.layoutSize - itemInfo.size
+    }
+
+    /**
+     * Ideally this would exist on [LazyListLayoutInfo] but it doesn't right now.
+     * Raised https://issuetracker.google.com/issues/200920410 to track.
+     */
+    private val LazyListLayoutInfo.layoutSize: Int
+        get() = if (this is LazyListLayoutInfoWithContentPadding) {
+            // If the LazyListLayoutInfo is actually a LazyListLayoutInfoWithContentPadding,
+            // use it's layoutSize
+            this.layoutSize
+        } else {
+            // This shouldn't happen through the usage of SnappingFlingBehavior, but otherwise
+            // we fallback to the end offset. That value contains the end content padding though.
+            viewportEndOffset
+        }
+}
+
+/**
+ * A snapping [FlingBehavior] for [LazyListState]. Typically this would be created
+ * via [rememberSnappingFlingBehavior].
+ *
+ * Note: the default parameter value for [decayAnimationSpec] is different to the value used in
```

Comment:
I would prefer to not provide a default value instead for this param

Reply:
Done.